Dockerfile

```
@@ -28,41 +28,87 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash - \
  && apt-get install -y nodejs
 
-# Downloading dependencies, these should be pinned to specific versions.
+# Add dependencies. All should be pinned to specific versions, except
+# Nextstrain-maintained software.
 # This includes pathogen-specific workflow dependencies. Since we only maintain a
 # single Docker image to support all pathogen workflows, some pathogen-specific
 # functionality must live in this Dockerfile. The following dependencies may be
 # used by multiple pathogen workflows, but they have been commented according to
 # the original pathogen that added these dependencies.
 
-# mafft
-WORKDIR /build/mafft
-RUN curl -fsSL https://mafft.cbrc.jp/alignment/software/mafft-7.475-linux.tgz \
-  | tar xzvpf - --no-same-owner --strip-components=2 mafft-linux64/mafftdir/
+# Create directories to be copied in final stage.
+RUN mkdir -p /out/bin /out/share /out/libexec
+
+
+# 1. Build programs from source
 
-# RAxML
+# Build RAxML
+# AVX should be widely-supported enough
 WORKDIR /build/RAxML
 RUN curl -fsSL https://api.github.com/repos/stamatak/standard-RAxML/tarball/v8.2.12 \
-  | tar xzvpf - --no-same-owner --strip-components=1
-RUN make -f Makefile.AVX.PTHREADS.gcc   # AVX should be widely-supported enough
+  | tar xzvpf - --no-same-owner --strip-components=1 \
+ && make -f Makefile.AVX.PTHREADS.gcc \
+ && cp -p raxmlHPC-PTHREADS-AVX /out/bin
 
-# FastTree
+# Build FastTree
 WORKDIR /build/FastTree
 RUN curl -fsSL https://api.github.com/repos/tsibley/FastTree/tarball/50c5b098ea085b46de30bfc29da5e3f113353e6f \
-  | tar xzvpf - --no-same-owner --strip-components=1
-RUN make FastTreeDblMP
+  | tar xzvpf - --no-same-owner --strip-components=1 \
+ && make FastTreeDblMP \
+ && cp -p FastTreeDblMP /out/bin
 
-# IQ-TREE
-WORKDIR /build/IQ-TREE
-RUN curl -fsSL https://github.com/iqtree/iqtree2/releases/download/v2.1.2/iqtree-2.1.2-Linux.tar.gz \
-  | tar xzvpf - --no-same-owner --strip-components=1
-RUN mv bin/iqtree2 bin/iqtree
-
-# vcftools
+# Build vcftools
 WORKDIR /build/vcftools
 RUN curl -fsSL https://github.com/vcftools/vcftools/releases/download/v0.1.16/vcftools-0.1.16.tar.gz \
-  | tar xzvpf - --no-same-owner --strip-components=2
-RUN ./configure --prefix=$PWD/built && make && make install
+  | tar xzvpf - --no-same-owner --strip-components=2 \
+ && ./configure --prefix=$PWD/built \
+ && make && make install \
+ && cp -rp built/bin/*    /out/bin \
+ && cp -rp built/share/*  /out/share
```

Maybe this?

```suggestion
 && ./configure --prefix=/out \
 && make && make install
```

or maybe explicit is better.

---

This alternative hadn't crossed my mind, thanks for pointing it out. I do think explicit is better here, since otherwise it's hard to tell how `/final/share` is used. Will leave as-is.